core/src/main/java/org/sonarsource/sonarlint/core/local/only/XodusLocalOnlyIssueStore.java

```diff
@@ -0,0 +1,260 @@
+  private static final String RESOLUTION_DATE_PROPERTY_NAME = "resolvedDate";
+  private static final String RULE_KEY_PROPERTY_NAME = "ruleKey";
+  private static final String RANGE_HASH_PROPERTY_NAME = "rangeHash";
+  private static final String LINE_HASH_PROPERTY_NAME = "lineHash";
+  private static final String START_LINE_PROPERTY_NAME = "startLine";
+  private static final String START_LINE_OFFSET_PROPERTY_NAME = "startLineOffset";
+  private static final String END_LINE_PROPERTY_NAME = "endLine";
+  private static final String END_LINE_OFFSET_PROPERTY_NAME = "endLineOffset";
+  private static final String MESSAGE_BLOB_NAME = "message";
+  private static final String BACKUP_TAR_GZ = "local_only_issue_backup.tar.gz";
+  private final PersistentEntityStore entityStore;
+  private final Path backupFile;
+  private final Path xodusDbDir;
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  public XodusLocalOnlyIssueStore(Path backupDir, Path workDir) throws IOException {
+    xodusDbDir = Files.createTempDirectory(workDir, "xodus-local-only-issue-store");
+    backupFile = backupDir.resolve(BACKUP_TAR_GZ);
+    if (Files.isRegularFile(backupFile)) {
+      LOG.debug("Restoring previous local-only issue database from {}", backupFile);
+      try {
+        TarGzUtils.extractTarGz(backupFile, xodusDbDir);
+      } catch (Exception e) {
+        LOG.error("Unable to restore local-only issue backup {}", backupFile);
+      }
+    }
+    LOG.debug("Starting local-only issue database from {}", xodusDbDir);
+    this.entityStore = buildEntityStore();
+    entityStore.executeInTransaction(txn -> {
+      entityStore.registerCustomPropertyType(txn, Instant.class, new InstantBinding());
+      entityStore.registerCustomPropertyType(txn, UUID.class, new UuidBinding());
+      entityStore.registerCustomPropertyType(txn, IssueStatus.class, new IssueStatusBinding());
+    });
+  }
+
+  public List<LocalOnlyIssue> load(String configurationScopeId, String filePath) {
+    return loadIssue(configurationScopeId, filePath, XodusLocalOnlyIssueStore::adapt);
+  }
+
+  private <G> List<G> loadIssue(String configurationScopeId, String filePath, Function<Entity, G> adapter) {
+    return entityStore.computeInReadonlyTransaction(txn -> findUnique(txn, CONFIGURATION_SCOPE_ID_ENTITY_TYPE, NAME_PROPERTY_NAME, configurationScopeId)
+      .map(branch -> branch.getLinks(CONFIGURATION_SCOPE_ID_TO_FILES_LINK_NAME))
+      .flatMap(files -> findUniqueAmong(files, PATH_PROPERTY_NAME, filePath))
+      .map(fileToLoad -> fileToLoad.getLinks(XodusLocalOnlyIssueStore.FILE_TO_ISSUES_LINK_NAME))
+      .map(issueEntities -> StreamSupport.stream(issueEntities.spliterator(), false)
+        .map(adapter)
+        .collect(Collectors.toList()))
+      .orElseGet(Collections::emptyList));
+  }
+
+  public void createLocalOnlyIssue(String configurationScopeId, LocalOnlyIssue issue, IssueStatus newStatus) {
```

Comment:
I would suggest `store` or `save` or something else than `create` which seems strange as we don't create it (it is a parameter)